storage/replica_test.go

```diff
@@ -487,6 +487,20 @@ func TestRangeLeaderLease(t *testing.T) {
 	if held, expired := hasLease(tc.rng, tc.clock.Now()); held || !expired {
 		t.Errorf("expected another replica to have expired lease")
 	}
+
+	// Verify that command returns NotLeaderError when lease is rejected.
+	tc.rng.proposeRaftCommandFn = func(id cmdIDKey, cmd roachpb.RaftCommand) <-chan error {
```

Comment:
if this range is part of a store (and it seems it is), then this could race since the store will periodically make use of its first range. You'd have to set it earlier (check how the other tests that use this mock are doing it).